client/core/core.go

```diff
@@ -6557,3 +6559,203 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
+				unitInfo := func(assetID uint32, symbol string) dex.UnitInfo {
+					assetInfo, err := asset.Info(assetID)
+					if err != nil {
+						xc, found := c.Exchanges()[cord.Host]
+						if !found {
+							return defaultUnitInfo(symbol)
+						}
+						asset, found := xc.Assets[assetID]
+						if !found || asset.UnitInfo.Conventional.ConversionFactor == 0 {
+							return defaultUnitInfo(symbol)
+						}
+					}
+					return assetInfo.UnitInfo
+				}
+
+				ordReader := &OrderReader{
+					Order:         cord,
+					BaseUnitInfo:  unitInfo(cord.BaseID, cord.BaseSymbol),
+					QuoteUnitInfo: unitInfo(cord.QuoteID, cord.QuoteSymbol),
+				}
+
+				timestamp := encode.UnixTimeMilli(int64(cord.Stamp)).Local().Format(time.RFC3339Nano)
+				err = csvWriter.Write([]string{
+					cord.Host,                     // Host
+					ord.Order.ID().String(),       // Order ID
+					cord.BaseSymbol,               // Base
+					cord.QuoteSymbol,              // Quote
+					ordReader.BaseQtyString(),     // Base Quantity
+					ordReader.SimpleRateString(),  // Order Rate
+					ordReader.AverageRateString(), // Actual Rate
+					ordReader.BaseAssetFees(),     // Base Fees
+					ordReader.QuoteAssetFees(),    // Quote Fees
+					ordReader.Type.String(),       // Type
+					ordReader.SideString(),        // Side
+					cord.TimeInForce.String(),     // Time in Force
+					ordReader.StatusString(),      // Status
+					ordReader.FilledPercent(),     // Filled
+					ordReader.SettledPercent(),    // Settled
+					timestamp,                     // Time
+				})
+				if err != nil {
+					return fmt.Errorf("error writing orders CSV: %v", err)
+				}
+				csvWriter.Flush()
+				err = csvWriter.Error()
+				if err != nil {
+					return fmt.Errorf("error writing orders CSV: %v", err)
+				}
+			}
+
+		}
+	}
+
+	// If provided a file to write the matches csv to, write the header and
+	// defer closing the file.
+	if matchesFile != nil {
+		defer matchesFile.Close()
+		csvWriter = csv.NewWriter(matchesFile)
+
+		csvWriter.UseCRLF = runtime.GOOS == "windows"
+		err = csvWriter.Write([]string{
+			"Host",
+			"Base",
+			"Quote",
+			"Match ID",
+			"Order ID",
+			"Quantity",
+			"Rate",
+			"Swap Fee Rate",
+			"Swap Address",
+			"Status",
+			"Side",
+			"Secret Hash",
+			"Maker Swap Coin ID",
+			"Maker Redeem Coin ID",
+			"Taker Swap Coin ID",
+			"Taker Redeem Coin ID",
+			"Refund Coin ID",
+			"Time",
+		})
+		if err != nil {
+			return fmt.Errorf("error writing matches CSV: %v", err)
+		}
+		csvWriter.Flush()
+		err = csvWriter.Error()
+		if err != nil {
+			return fmt.Errorf("error writing matches CSV: %v", err)
+		}
+	}
+
+	// Delete matches while saving to csv if available until the database
+	// says that's all or context is canceled.
+	finished = false
+	for !finished {
+		if err := c.ctx.Err(); err != nil {
+			return fmt.Errorf("unable to delete matches: %v", err)
+		}
+		var mtchs []*db.MetaMatch
+		mtchs, finished, err = c.db.DeleteInactiveMatches(olderThan)
```

Comment:
In case there's an error somewhere, here or in the orders loop above, I wonder if the DB would be in a better state if we deleted matches before deleting orders.  Maybe it doesn't matter.  Thoughts?

Reply:
Can delete matches first.

Reply:
Deleting matches first.